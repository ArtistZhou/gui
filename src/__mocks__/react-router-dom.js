import React from 'react';
import { compose, setDisplayName } from 'recompose';
import * as rrd from 'react-router-dom';
// Just render plain div with its children
rrd.BrowserRouter = compose(setDisplayName('BrowserRouter'))(({ children }) => <div>{children}</div>);

module.exports = rrd;
