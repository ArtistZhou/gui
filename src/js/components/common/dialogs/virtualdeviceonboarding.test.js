import React from 'react';
import { MemoryRouter } from 'react-router-dom';
import renderer from 'react-test-renderer';
import { Provider } from 'react-redux';
import configureStore from 'redux-mock-store';
import VirtualDeviceOnboarding from './virtualdeviceonboarding';

const mockStore = configureStore([]);
const store = mockStore({
  app: { features: { isHosted: false } },
  users: {
    organization: { tenant_token: null }
  }
});
store.dispatch = jest.fn();

it('renders correctly', () => {
  const tree = renderer
    .create(
      <MemoryRouter>
        <Provider store={store}>
          <VirtualDeviceOnboarding />
        </Provider>
      </MemoryRouter>
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});
