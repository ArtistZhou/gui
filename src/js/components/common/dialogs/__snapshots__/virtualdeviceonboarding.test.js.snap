// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div>
  <div>
    <b>
      1. Prerequisites
    </b>
    <p>
      As you are running Mender on-premise, for these instructions we assume that you already have Docker installed and the Mender integration environment up and running on your machine.
    </p>
    <p>
      To start a virtual device, change directory into the folder where you cloned Mender integration.
    </p>
  </div>
  <p>
    <b>
      2. Copy & paste and run the following command to start the virtual device:
    </b>
  </p>
  <div
    className="code"
  >
    <button
      className="MuiButtonBase-root-27 MuiButton-root-1 MuiButton-text-3 MuiButton-flat-6"
      disabled={false}
      icon={
        <WithStyles(Icon)
          className="material-icons"
        >
          content_paste
        </WithStyles(Icon)>
      }
      onBlur={[Function]}
      onClick={[Function]}
      onContextMenu={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      onKeyUp={[Function]}
      onMouseDown={[Function]}
      onMouseLeave={[Function]}
      onMouseUp={[Function]}
      onTouchEnd={[Function]}
      onTouchMove={[Function]}
      onTouchStart={[Function]}
      style={
        Object {
          "float": "right",
          "margin": "-10px 0 0 10px",
        }
      }
      tabIndex="0"
      type="button"
    >
      <span
        className="MuiButton-label-2"
      >
        Copy to clipboard
      </span>
      <span
        className="MuiTouchRipple-root-30"
      />
    </button>
    <span
      style={
        Object {
          "wordBreak": "break-word",
        }
      }
    >
      ./demo --client up
    </span>
  </div>
  <p />
  <p>
    The device should appear in the Pending devices view in a couple of minutes.
  </p>
  <p>
    Visit the 
    <a
      href="/help/application-updates/demo-virtual-device"
      onClick={[Function]}
    >
      Virtual Devices Help page
    </a>
     for more info on managing the virtual device.
  </p>
</div>
`;
